### ISSUE TITLE
Implement TS cluster missile

### ISSUE THREAD

**Issue by pchote** [OpenRA/OpenRA#7773]
It is currently implemented as a placeholder.

**Comment by Mailaender** [comment]
I found https://github.com/GraionDilach/OpenRA.Mods.AS/blob/master/Traits/Warheads/FireClusterWarhead.cs

**Comment by GraionDilach** [comment]
I'd gladly shift that to upstream, but as it stands ^ isn't fine for the TS Cluster Missile because TS missile had a 3x3 rectangle while that one (and the underlying OpenRA functions I could find) only support radii.

**Comment by Mailaender** [comment]
We could use it for the Dune 2000 Death Hand missile in the mean time.

**Comment by penev92** [comment]
FWIW I prefer a circular spread over rectangular, it makes much more sense. I bet the rectangular spread in TS was a result of engine/time limitations.

**Comment by GraionDilach** [comment]
The D2k Death Hand is ANOTHER thingy, which I should find time to implement for once (that seems to be using the same logic as say Prism Tank shrapnel from what I seen).

@penev92 I'd prefer having both.

**Comment by StormWingDelta** [comment]
hmm maybe implement a Cluster Warhead for things like the Cluster Missile? Than have some detonation logic for when it should go off, before target, on target, etc.  The Cluster Warhead would spawn another projectile as its armament at some variable amount and randomness.

Just an idea but might help.

**Comment by GraionDilach** [comment]
@StormWingDelta Yes, that's exactly what I did months ago, but my solution requires a dependency due to mod requirements.

**Comment by reaperrr** [comment]
I PR'd the warhead (with minor changes) in #16363.
